ui/app/models/secret-engine.js

```diff
@@ -108,15 +108,8 @@ export default class SecretEngineModel extends Model {
   deleteVersionAfter;
 
   /* GETTERS */
-  get modelTypeForKV() {
-    const engineType = this.engineType;
-    if ((engineType === 'kv' || engineType === 'generic') && this.version === 2) {
-      return 'secret-v2';
-    }
-    return 'secret';
-  }
   get isV2KV() {
-    return this.modelTypeForKV === 'secret-v2';
+    return this.engineType === 'kv' && this.version === 2;
```

Comment:
```suggestion
    return (this.version === 2 && (this.engineType === 'kv' || this.engineType === 'generic'));
```
We do need to keep the generic check, because generic is basically the same thing as kv1, and you can upgrade it to a "version 2" but it keeps its type as "generic". For illustration: 

* Open vault UI web CLI
* Mount a generic mount `write sys/mounts/generic type=generic`
* Upgrade it to v2 `write sys/mounts/generic/tune options=version=2`

And then you have a v2 generic engine 🙃

Reply:
Wow! There it is! I couldn't figure out how'd we ever get generic and this.version 2 together. Thank you! I'll add a comment about this for future.